### PR TITLE
signal polarity configuration implemented and tested

### PR DESCRIPTION
**Pull request by basaks**
@mcgarth edit:
This PR implements a simple signal polarity conversion, which is applied to output geotiff products generated during the MERGE step. Data sign is not affected in the other processing steps. Default is multiplying the output data arrays by -1. This ensures GA's standard input interferograms are converted such that PyRate outputs have range increases as negative and range decreases as positive.
- Metadata for `signal_polarity` and `velerror_nsig` has been added to the geotiffs
- Info messages have been added to MERGE step to report on conversions being applied

### CODE REVIEW

File: input_parameters.conf [hunk 1]
```diff
@@ -103,6 +103,13 @@ noDataValue:   0.0
 # Nan conversion flag. Set to 1 if missing No-data values are to be converted to NaN
 nan_conversion: 1
 
+
+# sign convention for phase data
```
Comment (richardt94) [inline input_parameters.conf:107]: Correct comments - subsidence as negative and uplift as positive mean the same thing.

Also maybe add a disclaimer that the sign convention only works as stated if inputs are in standard GAMMA format?

File: tests/test_merge.py [hunk 2]
```diff
@@ -96,54 +102,74 @@ def test_los_conversion_comparison(self):
...
             k_dir = Path(params[C.OUT_DIR]).joinpath(ifc.LOS_PROJECTION_OPTION[k])
             k_dir.mkdir(exist_ok=True)
-
-            for out_type in los_projection_out_types:
-                for tif in Path(params[C.OUT_DIR]).glob(out_type + '*.tif'):
-                    shutil.move(tif, k_dir.joinpath(tif.name))
             all_dirs[k] = k_dir
+            self.run_with_new_params(k_dir, params)
+
+        signal_dir = Path(params[C.OUT_DIR]).joinpath('signal_polatiry_dir')
```
Comment (richardt94) [inline tests/test_merge.py:113]: Typo, polatiry -> polarity

File: pyrate/merge.py [hunk 3]
```diff
@@ -310,7 +310,7 @@ def __save_merged_files(ifgs_dict, params, array, out_type, index=None, savenpy=
                          f"{ifc.LOS_PROJECTION_OPTION[params[C.LOS_PROJECTION]]} for file {dest}")
             incidence = shared.Geometry(incidence_path)
             incidence.open()
-            array /= los_projection_divisors[params[C.LOS_PROJECTION]](incidence.data)
+            array /= los_projection_divisors[params[C.LOS_PROJECTION]](incidence.data) * params[C.SIGNAL_POLARITY]
```
Comment (richardt94) [inline pyrate/merge.py:313]: The sign convention is applied in merge, but in the example configs it is listed as a parameter for prepifg

**Review by richardt94 — APPROVED** [review]
Looks good to me. Some changes/explanation needed in the config file comments as noted.

File: input_parameters.conf (hunk 1 above)
Comment (mcgarth) [reply input_parameters.conf:107]: N.B. there is no "standard" GAMMA format ;-)

Comment (richardt94) [reply input_parameters.conf:107]: Well in that case maybe we want something like this?
```
+1 - keep sign convention of input interferograms
-1 - reverse sign convention compared to input interferograms
```
We don't want a user putting in interferograms where negative values already correspond to motion away from the satellite, and expecting `signal_polarity: -1` to preserve that.

Comment (mcgarth) [reply input_parameters.conf:107]: Yes great point. We still need our user to understand what their input data signal convention is.

File: pyrate/merge.py (hunk 3 above)
Comment (mcgarth) [reply pyrate/merge.py:313]: I have removed the extra comments from backend test configs, and updated the top level config to show it is applied in MERGE

File: input_parameters.conf (hunk 1 above)
Comment (mcgarth) [reply input_parameters.conf:107]: I have updated `input_parameters.conf` based on @richardt94's idea

**Review by mcgarth — APPROVED** [review]
I made some additions. @richardt94 to re-review please

**Review by richardt94 — APPROVED** [review]
All looks good to me. Velocity error sigma values are now included in the geotiff metadata which is a nice addition too.

**Pull request merged**